### ISSUE TITLE
Could not resolve dependency with React 18

### ISSUE THREAD

**Issue by AnastasiiaBardyuzha** [gregnb/mui-datatables#1916]
I tried to install mui-datatables. And I caught error:

npm ERR! Found: react@18.0.0
npm ERR! node_modules/react
npm ERR!   react@"18.0.0" from the root project
npm ERR!   peer react@">=16.8.0" from @emotion/react@11.9.0
npm ERR!   node_modules/@emotion/react
npm ERR!     @emotion/react@"11.9.0" from the root project
npm ERR!     peer @emotion/react@"^11.4.1" from mui-datatables@4.2.2
npm ERR!     node_modules/mui-datatables
npm ERR!       mui-datatables@"*" from the root project
npm ERR!     2 more (@mui/material, @emotion/styled)
npm ERR!   4 more (@mui/icons-material, @mui/material, @emotion/styled, react-dom)
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^16.8.0 || ^17.0.2" from mui-datatables@4.2.2
npm ERR! node_modules/mui-datatables
npm ERR!   mui-datatables@"*" from the root project 
 

| Tech         | Version |
|--------------|---------|
| Material-UI  |    5.5.2     |
| MUI-datatables  |     4.2.2    |
| React        |   18.0.0      |

**Comment by fabianmeneses56** [comment]
same problem

**Comment by jasonwee** [comment]
same problem too

**Comment by dayron9408** [comment]
same problem too

**Comment by dayron9408** [comment]
> same problem too

it workerd for me using npm install mui-datatables --save --legacy-peer-deps

**Comment by dayron9408** [comment]
> same problem

npm install mui-datatables --save --legacy-peer-deps

**Comment by parzhitsky** [comment]
Same problem for me, unfortunately.

@dayron9408 `--legacy-peer-deps` can potentially lead to inconsistent dependency tree:

```sh
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

**Comment by thetaPC** [comment]
Are there plans to upgrade to React 18?

**Comment by Marko351** [comment]
I have the same problem. I'll use --legacy-peer-deps to start a project but I hope this issue will be solved in the future release.

[https://ibb.co/SxGdzsq](url)

**Comment by TalhaAmjad0034** [comment]
Same issue, on deployment...

**Comment by dmachibya** [comment]
same problem

**Comment by danieljmann96** [comment]
Same issue, are you planning a react 18 upgrade?

**Comment by TalhaAmjad0034** [comment]
I've changed my version to 17.0.2 and that error has resolved.

**Comment by ShetGanesh** [comment]
Facing the same issue

**Comment by Kristiyan96** [comment]
Me too

**Comment by MiKaminskas** [comment]
same here

**Comment by Abam** [comment]
Facing same issue with React 18. 
Any plan to upgrade the dependency tree of this package ?

**Comment by ronbeilinson** [comment]
same!

**Comment by mvandoff-avi** [comment]
Same, please upgrade! <3

**Comment by wickedvic** [comment]
Bump ^^

**Comment by harryd31** [comment]
running into the same issue

**Comment by shafiq-techmentors** [comment]
same issue. please upgrade

**Comment by Eliot27** [comment]
```
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

i need some solution

**Comment by ashwin-07** [comment]
Hey! Facing same issue while using React 18, with `mui-datatables: 4.2.2`
using  ` npm install --legacy-peer-deps` for now. Any plans to upgrade?

**Comment by MuLoo** [comment]
if anyone find this,  upgrade `mui-datatables` to `4.3.0` will be ok

**Comment by johnpipi** [comment]
> if anyone find this, upgrade `mui-datatables` to `4.3.0` will be ok

nope i have mui-datatables: 4.3.0 with mui: 5.11.13

Same problem

**Comment by gbowne1** [comment]
yeah I have the same problem.. only noticed this when I tried to sandbox my app to figure out some of its problems. 

does this only work with React 17.0.0 ~ 17.0.2?

**Comment by jelsabird** [comment]
Same problem

**Comment by stefanus-ricky** [comment]
Still a problem, seems all MUI library not using React 18 not just the datatable. The table still work though, haven't see any issue beside installation so far.
